internal/resolution/variablesources/bundle_deployment.go

```diff
@@ -36,20 +42,15 @@ func (o *BundleDeploymentVariableSource) GetVariables(ctx context.Context) ([]de
 		return nil, err
 	}
 
-	processed := map[string]struct{}{}
 	for _, bundleDeployment := range bundleDeployments.Items {
-		sourceImage := bundleDeployment.Spec.Template.Spec.Source.Image
-		if sourceImage != nil && sourceImage.Ref != "" {
-			if _, ok := processed[sourceImage.Ref]; ok {
-				continue
-			}
-			processed[sourceImage.Ref] = struct{}{}
-			ips, err := NewInstalledPackageVariableSource(o.catalogClient, bundleDeployment.Spec.Template.Spec.Source.Image.Ref)
-			if err != nil {
-				return nil, err
-			}
-			variableSources = append(variableSources, ips)
+		pkgName := bundleDeployment.Annotations[AnnotationPackageName]
+		bundleName := bundleDeployment.Annotations[AnnotationBundleName]
+		bundleVersion := bundleDeployment.Annotations[AnnotationBundleVersion]
+		if pkgName == "" || bundleName == "" || bundleVersion == "" {
+			continue
```

Comment:
Do we need to care about bundle deployments created directly? These annotations will only be present on bundle deployments reconciled by operator-controller.

Reply:
For now, I don't think we need to care, at least in this context.

I sorta think this context is: is there a BD _for this operator_ already? (And maybe that means we should have a label on the BD like `operators.operatorframework.io/operator-name` whose value is `Operator.metadata.name`?

Perhaps in the future, there's a separate feature that helps the resolver become aware of other BDs, but I don't think we need to do that now.